### DISCUSSION TITLE
Live Looping Midievents

### DISCUSSION THREAD

**Discussion by lleck** [monocasual/giada#489] [Q&A]
Hi All,
First of all I want to thank everyone who contributed to monocausal for developing this great piece of software.
trying different sequencers on my raspi 4 equipped with a 7" screen, giada turned out to be the perfectly scaled solution. everything fits the screen and with the ease of the controller setup via midi learning I can quickly develop a live setup.
However I am having troubles recording midi into a track. No matter if I use Alsa or Jack as the driver I am not able to record any events into an armed midi track. Output of hand-drawn midi events works as expected. 
Being able to liveloop the midi data that I put into giada via a midikeyboard or drumpad would be the coolest thing, but as for now I am not 100% sure if this feature is even implemented yet. ?! 

P.S.: I am running Giada 0.18.1 built from source on a raspi 4b with patchboxOS by blokas.io

thanks in advance,
keep up the good work!
Lukas

**Comment by gvnnz** [comment]
Hey Lukas, thanks a lot for your kind words!  

The live MIDI recording feature has been implemented a while ago. There is a known bug though (#482) we are going to fix in the next version. That said, you definitely should be able to record live MIDI: could you try to enable logging and see if you are receving MIDI messages in the first place? 

How to enable logging: Config -> Misc -> Debug Messages -> To standard output

Thanks!

**Reply by lleck** [reply]
Hi gvnnz,

I will give it another try next week and enable logging to check wether my midi gets through! I 
I am effectively using a nanokontrol to operate the transport and individual loops, maybe my problem is more alsa-midi related, since I'm  having sorrows there whith individual ports of my midi interface being shown as a one port device only, however giada shows all existing ports in the midi settings. 

Thanks for your help ;)

**Reply by lleck** [reply]
Hi, after several sidetracks I managed to get midi recorded this time. I'm not sure what went wrong when I first tried. However on my system ( raspi 4b) there is still an issue: The midi output won't mute when I activate the mute button. In general ( I think this has been discussed in the general midi overhaul thread) it would be beneficial to enable midi device/port selection on each midi channel for both in and output. That way users like me could benefit from hardware midi interfaces with several ports allowing to adress more than 16 channels. Ideally these would show up as jack or alsa ports to be patched in patchage or catia. Alternatively or additionally an extensive menu for midi-filtering could let the user select different message types from different ports directly in giada. 
So far my experience with midi. I will keep on using giada, and hope to contribute with my user experiences on arm architecture. ✌️
thanks again to all developers

**Reply by gvnnz** [reply]
@lleck thanks for your feedback! Can't test MIDI output right now, but by looking at the source code I can definitely tell the MIDI output doesn't mute on... mute. I've opened a new issue for that (#497). 

Regarding multiple MIDI devices: yes, that's the idea, both for MIDI and audio, so that each channel can send/receive data from/to specific devices. Being a overhaul, I guess such feature will be included after v1.0 is out. 

Thank you again for your help,  hope to hear from you soon :rocket:

**Reply by CarloCattano** [reply]
Hey , runing Giada on a pi 3 here too :-)  I can record , playback and all. The only thing is I cant hear the notes I play if the track is armed or playing but Master recording is not active . Im using midi keyboard -> Giada -> midithrough -> Carla lv2 synth plguin

**Reply by lleck** [reply]
Hi, great to hear more people are enjoying giada on raspi! I cannot emphasize enough how well it fits the pi with a 7" screen. Thanks @gvnnz for issuing the midi mute problem. I hope I'll get back to my rack soon to play a little more with giada ...
